src/effects/effects_base/internal/effectsuiactions.cpp

```diff
@@ -95,6 +90,32 @@ void EffectsUiActions::init()
         it->title = REPEAT_LAST_EFFECT_TITLE.arg(effectTitle);
         m_actionsChanged.send({ *it });
     });
+
+    auto makeActions = [this](const EffectMetaList& effects) {
+        m_actions.clear();
+        m_actions.reserve(effects.size() + STATIC_ACTIONS.size());
+
+        for (const EffectMeta& e : effects) {
+            UiAction action;
+            action.code = makeEffectOpenAction(e.id).toString();
+            action.uiCtx = context::UiCtxProjectOpened;
+            action.scCtx = context::CTX_PROJECT_FOCUSED;
```

Comment:
Why do `uiCtx` and `scCtx` differ ? If `action.uiCtx = context::UiCtxProjectOpened;`, I'd have expected `action.scCtx = context::CTX_PROJECT_OPENED;`

Reply:
This is a simple and complex topic at the same time :) 

With the Ui context, everything is quite simple - it determines when an action is available depending on which page is open. For example, we might be on the home screen, or we might be in the project screen (with a project loaded)... the current UI context tells us where we are now. 

Shortcut context is a little different. It tells when we can accept the shortcut of this action.
The whole problem is that the number of shortcuts, key combinations, is very limited. Therefore, a situation arises when one and the same shortcut must trigger different actions, depending on where exactly the user is.

Let's assume that we have some panel on the project page, for example a mixer or something else. And there are some actions... and we can assign a shortcut to them... and this shortcut will match the shortcut assigned to the effect, what should we do with this?
Options:
1. Do not allow assigning the same shortcut to different actions - but there are too few keyboard shortcuts..
2. We can determinate where exactly the user is now and, accordingly, what exactly he expects when this shortcut is triggered. If the user clicks on clips and triggers a shortcut to which the effect is assigned - we need to show the form for working with the effect. If the user clicks on the mixer and triggers the same shortcut, then we need to call the action associated with the mixer (for example)

At the same time, the actions themselves can be available regardless of which panel the user is currently on (they can simply be sent in some other way, for example by clicking on the UI control)

Reply:
Ok, I stupidly imagined `uiCtx` and `scCtx` were duplication of a same concept. Thanks for the clarification, will be useful soon enough :D